.github/workflows/release-with-changeset.yml

```diff
@@ -0,0 +1,141 @@
+# This action is centrally managed in https://github.com/asyncapi/.github/
+# Don't make changes to this file in this repo as they will be overwritten with changes made to the same file in the above-mentioned repo
+
+# It does magic only if there is a package.json file in the root of the project
+name: Release with changeset
+
+on:
+  push:
+    branches:
+      - master
+      # The below lines are not enough to have release supported for these branches
+      # Make sure the configuration of the `semantic-release` package mentions these branches
+      - next-spec
+      - next-major
+      - next-major-spec
+      - beta
+      - alpha
+      - next
```

Comment:
Do we need anything else to be configured (perhaps in changeset config file?) to allow those branches to be released as well? How does it work?

Reply:
I found out this in their docs: https://github.com/changesets/changesets/blob/main/docs/prereleases.md

It should be done manually it seems. I guess in the workflow, detecting we are not in master branch and run `npx changeset pre enter {tag}` then. More investigation is needed. 

We can work on it in a second iteration.






[12:08](https://asyncapi.slack.com/archives/D03CPF9EXQF/p1718878116286129)
Seems like npx changeset pre enter {tag}

Reply:
ok sure will do that.